langchain/src/tools/tests/ical.test.ts

```diff
@@ -0,0 +1,84 @@
+import { it, expect, jest } from "@jest/globals";
+
+import { BaseLLM, LLM } from "../../llms/base.js";
+import { BaseFileStore } from "../../schema/index.js";
+
+import { ICalTool } from "../ical.js";
+import {
+  getIcsString,
+  getInputCommand,
+  getOutputPrompt,
+} from "../fixtures/ical.js";
+
+export class FakeLLM extends LLM {
+  _llmType(): string {
+    return "fake";
+  }
+
+  async _call(prompt: string): Promise<string> {
+    return prompt;
+  }
+}
+
+describe("ICalTool", () => {
+  let tool: ICalTool;
+  let store: BaseFileStore;
```

Comment:
Could we make this a tool for a given provider (e.g. Google Calendar, Outlook) instead?

A little bit uncertain about allowing tools to rely on the existence of a filesystem, and am not certain about the utility of storing these locally anyway.